### PR TITLE
login:use PATH_RUN_SYSTEMD_SEATS instead of /run/systemd/seats

### PR DESCRIPTION
**Pull request by JackFangXN**
use PATH_RUN_SYSTEMD_SEATS instead of /run/systemd/seats, code looks better.

### CODE REVIEW

File: src/libsystemd/sd-login/sd-login.c [hunk 1]
```diff
@@ -250,7 +250,7 @@ static int file_of_uid(uid_t uid, char **p) {
         assert_return(uid_is_valid(uid), -EINVAL);
         assert(p);
 
-        if (asprintf(p, "/run/systemd/users/" UID_FMT, uid) < 0)
+        if (asprintf(p, PATH_RUN_SYSTEMD_USERS UID_FMT, uid) < 0)
```
Comment (yuwata) [inline src/libsystemd/sd-login/sd-login.c]: Needs "/" between `PATH_RUN_SYSTEMD_USERS` and `UID_FMT`.

File: src/login/logind-user.c [hunk 2]
```diff
@@ -63,7 +63,7 @@ int user_new(User **ret,
                 .last_session_timestamp = USEC_INFINITY,
         };
 
-        if (asprintf(&u->state_file, "/run/systemd/users/" UID_FMT, ur->uid) < 0)
+        if (asprintf(&u->state_file, PATH_RUN_SYSTEMD_USERS UID_FMT, ur->uid) < 0)
```
Comment (yuwata) [inline src/login/logind-user.c]: Same here.

File: src/login/logind-session.c [hunk 3]
```diff
@@ -1066,11 +1066,11 @@ int session_create_fifo(Session *s) {
 
         /* Create FIFO */
         if (!s->fifo_path) {
-                r = mkdir_safe_label("/run/systemd/sessions", 0755, 0, 0, MKDIR_WARN_MODE);
+                r = mkdir_safe_label(PATH_RUN_SYSTEMD_SESSIONS, 0755, 0, 0, MKDIR_WARN_MODE);
                 if (r < 0)
                         return r;
 
-                s->fifo_path = strjoin("/run/systemd/sessions/", s->id, ".ref");
+                s->fifo_path = strjoin(PATH_RUN_SYSTEMD_SESSIONS, s->id, ".ref");
```
Comment (yuwata) [inline src/login/logind-session.c]: Same here. Missing "/".

File: src/login/logind-inhibit.c [hunk 4]
```diff
@@ -46,7 +46,7 @@ int inhibitor_new(Inhibitor **ret, Manager *m, const char* id) {
                 .fifo_fd = -1,
         };
 
-        i->state_file = path_join("/run/systemd/inhibit", id);
+        i->state_file = path_join(PATH_RUN_SYSTEMD_INHIBIT1, id);
```
Comment (yuwata) [inline src/login/logind-inhibit.c]: INHIBIT1???

Comment (JackFangXN) [reply src/login/logind-inhibit.c]: no, INHIBIT, input error. i find it and already fix it. thank you for your suggestion.

**Comment by JackFangXN** [comment]
@yuwata hi, i patch this patchs in local and can be success.  i don't know why CI is fail.  Please review my commits, is it OK,

**Comment by yuwata** [comment]
LGTM now. The bionic CIs are now broken let's wait for a moment and restart them later.

**Comment by poettering** [comment]
uh, i am not a fan of this. This adds an extra layer of indirection making things more opaque, the defines are longer than the literals even!

`#define`s are a great concept, but only if they either:

1. can be defined differently on different builds
2. are drastically shorter/easier to type/less likely to mistype then the literal versions the resolve to
3. arbitrary numeric values whose value is of no immediate relevance to the algorithm

But these paths don't qualify for any of this. You just replace "/foo/bar/baz" by PATH_FOO_BAR_BAZ, i.e. swap a few characters around, that's all...

I am sorry, not a fan.

**Comment by JackFangXN** [comment]
> uh, i am not a fan of this. This adds an extra layer of indirection making things more opaque, the defines are longer than the literals even!
> 
> `#define`s are a great concept, but only if they either:
> 
> 1. can be defined differently on different builds
> 2. are drastically shorter/easier to type/less likely to mistype then the literal versions the resolve to
> 3. arbitrary numeric values whose value is of no immediate relevance to the algorithm
> 
> But these paths don't qualify for any of this. You just replace "/foo/bar/baz" by PATH_FOO_BAR_BAZ, i.e. swap a few characters around, that's all...
> 
> I am sorry, not a fan.

en, you are right, i will abandon these patchs and try #define function to adapt different path.

**Pull request closed without merging**